### PR TITLE
FOGL-7814: Move readings in south plugin ingest.

### PR DESCRIPTION
**Pull request by cyberwalk3r**
- Added function to transfer readings vector in reading_set class.
- Added function to remove and return a reading from reading vector in reading_set class.
- Changed plugin ingest for poll and async using moveAllReadings function added to reading_set class.

### CODE REVIEW

**Review by AmandeepArora — COMMENTED** [review]
Looks good to me. It would help to check with valgrind with south/north service and filters.

File: C/common/reading_set.cpp [hunk 1]
```diff
@@ -306,6 +306,36 @@ ReadingSet::clear()
 	m_readings.clear();
 }
 
+/**
+ * Remove readings from the vector and return a reference to new vector
+ * containing readings*
+*/
+std::vector<Reading*>* ReadingSet::moveAllReadings()
+{
+	std::vector<Reading*>* transferredPtr = new std::vector<Reading*>(std::move(m_readings));
```
[the remark this answers is not in the document]
Comment (cyberwalk3r) [reply C/common/reading_set.cpp:315]: Okay, I'll make the suggested change.

**Review by MarkRiddoch — COMMENTED** [review]
I think it would be good to do a run of this under valgrind. You should be able to pick any of the async south plugins and give it a go.

**Comment by cyberwalk3r** [comment]
> I think it would be good to do a run of this under valgrind. You should be able to pick any of the async south plugins and give it a go.

@MarkRiddoch Attached log file for async csv plugin ran under valgrind. Please see the logs and if we can move ahead with this.
[south_valgrind.log](https://github.com/fledge-iot/fledge/files/12047900/south_valgrind.log)

**Comment by AmandeepArora** [comment]
> > I think it would be good to do a run of this under valgrind. You should be able to pick any of the async south plugins and give it a go.
> 
> @MarkRiddoch Attached log file for async csv plugin ran under valgrind. Please see the logs and if we can move ahead with this. [south_valgrind.log](https://github.com/fledge-iot/fledge/files/12047900/south_valgrind.log)

Hi @cyberwalk3r, can you please clarify what all user-cases you ran and how many readings were processed while valgrind was attached to this south service? The file doesn't seem to have enough/complete logs for Reading and Datapoint classes.

Maybe use following flags for valgrind so that all child processes are tracked "valgrind -v --leak-check=full --show-leak-kinds=all --track-origins=yes --tool=memcheck".

**Comment by cyberwalk3r** [comment]
> > > I think it would be good to do a run of this under valgrind. You should be able to pick any of the async south plugins and give it a go.
> > 
> > 
> > @MarkRiddoch Attached log file for async csv plugin ran under valgrind. Please see the logs and if we can move ahead with this. [south_valgrind.log](https://github.com/fledge-iot/fledge/files/12047900/south_valgrind.log)
> 
> Hi @cyberwalk3r, can you please clarify what all user-cases you ran and how many readings were processed while valgrind was attached to this south service? The file doesn't seem to have enough/complete logs for Reading and Datapoint classes.
> 
> Maybe use following flags for valgrind so that all child processes are tracked "valgrind -v --leak-check=full --show-leak-kinds=all --track-origins=yes --tool=memcheck".

The logs attached above are run with the mentioned options, test was run for csv-async plugin for 1 hour with only south service running.

**Comment by AmandeepArora** [comment]
@MarkRiddoch : Please advise whether the coverage & valgrind logs look good.

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**